### PR TITLE
NR-286847 Add fips compliant pkgs

### PR DESCRIPTION
**Pull request by sairaj18**
Add fips compliant package building.

[Ref PR](https://github.com/newrelic/nri-redis/pull/208)

[Test pre-release](https://github.com/newrelic/nri-mssql/actions/runs/12366888043)
`goreleaser` output with local testing
```shell
root@30d9d08c4004:/app# bin/goreleaser release --config ./build/.goreleaser.yml --clean --snapshot
  • skipping announce, publish and validate...
  • cleaning distribution directory
  • loading environment variables
  • getting and validating git state
    • git state                                      commit=98486bb5dee04b529d2fd2455dca362800447335 branch=master current_tag=v2.14.0 previous_tag=v2.13.0 dirty=true
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
    • pipe skipped                                   reason=release is disabled
  • snapshotting
    • building snapshot...                           version=2.14.0-SNAPSHOT-98486bb
  • ensuring distribution directory
  • setting up metadata
  • writing release metadata
  • loading go mod information
  • build prerequisites
  • building binaries
    • building                                       binary=dist/nri-nix_linux_amd64_v1/nri-mssql
    • building                                       binary=dist/nri-nix_darwin_arm64_v8.0/nri-mssql
    • building                                       binary=dist/nri-nix-fips_linux_arm64_v8.0/nri-mssql
    • building                                       binary=dist/nri-nix_linux_386_sse2/nri-mssql
    • building                                       binary=dist/nri-nix_linux_arm64_v8.0/nri-mssql
    • building                                       binary=dist/nri-nix_linux_arm_6/nri-mssql
    • building                                       binary=dist/nri-nix-fips_linux_amd64_v1/nri-mssql
    • building                                       binary=dist/nri-nix_darwin_amd64_v1/nri-mssql
  • archives
    • creating                                       archive=dist/nri-mssql-fips_linux_2.14.0-SNAPSHOT-98486bb_arm64_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql_linux_2.14.0-SNAPSHOT-98486bb_amd64_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql_linux_2.14.0-SNAPSHOT-98486bb_386_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql_linux_2.14.0-SNAPSHOT-98486bb_arm_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql_darwin_2.14.0-SNAPSHOT-98486bb_arm64_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql-fips_linux_2.14.0-SNAPSHOT-98486bb_amd64_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql_darwin_2.14.0-SNAPSHOT-98486bb_amd64_dirty.tar.gz
    • creating                                       archive=dist/nri-mssql_linux_2.14.0-SNAPSHOT-98486bb_arm64_dirty.tar.gz
  • linux packages
    • creating                                       package=nri-mssql format=deb arch=arm64v8.0 file=dist/nri-mssql_2.14.0-SNAPSHOT-98486bb-1_arm64.deb
    • creating                                       package=nri-mssql format=deb arch=arm6 file=dist/nri-mssql_2.14.0-SNAPSHOT-98486bb-1_arm.deb
    • creating                                       package=nri-mssql format=deb arch=amd64v1 file=dist/nri-mssql_2.14.0-SNAPSHOT-98486bb-1_amd64.deb
    • creating                                       package=nri-mssql format=deb arch=386sse2 file=dist/nri-mssql_2.14.0-SNAPSHOT-98486bb-1_386.deb
    • creating                                       package=nri-mssql format=rpm arch=386sse2 file=dist/nri-mssql-2.14.0-SNAPSHOT-98486bb-1.386.rpm
    • creating                                       package=nri-mssql format=rpm arch=arm64v8.0 file=dist/nri-mssql-2.14.0-SNAPSHOT-98486bb-1.arm64.rpm
    • creating                                       package=nri-mssql format=rpm arch=arm6 file=dist/nri-mssql-2.14.0-SNAPSHOT-98486bb-1.arm.rpm
    • creating                                       package=nri-mssql format=rpm arch=amd64v1 file=dist/nri-mssql-2.14.0-SNAPSHOT-98486bb-1.x86_64.rpm
    • creating                                       package=nri-mssql-fips format=rpm arch=arm64v8.0 file=dist/nri-mssql-fips-2.14.0-SNAPSHOT-98486bb-1.arm64.rpm
    • creating                                       package=nri-mssql-fips format=deb arch=arm64v8.0 file=dist/nri-mssql-fips_2.14.0-SNAPSHOT-98486bb-1_arm64.deb
    • creating                                       package=nri-mssql-fips format=deb arch=amd64v1 file=dist/nri-mssql-fips_2.14.0-SNAPSHOT-98486bb-1_amd64.deb
    • creating                                       package=nri-mssql-fips format=rpm arch=amd64v1 file=dist/nri-mssql-fips-2.14.0-SNAPSHOT-98486bb-1.x86_64.rpm
  • calculating checksums
  • writing artifacts metadata
  • release succeeded after 5s
  • thanks for using goreleaser!
``` 

Note: Commented the windows steps for local testing otherwise it fails with this error 
`⨯ release failed after 1m16s error=pre hook failed: template: failed to apply "build/windows/set_exe_properties.sh {{ .Env.TAG }} \"mssql\"": map has no entry for key "TAG"`

### CODE REVIEW

**Comment by sairaj18** [comment]
> Did you test it by creating a prerelease? I think you'll get an error when trying to sign the packages, because the files are not executable.
> 
> ```
> ❯ ll build/nix/
> total 40
> -rwxr-xr-x@ 1 rohanyadav  staff  1567 Nov 13 08:29 fix_archives.sh
> -rwxr-xr-x@ 1 rohanyadav  staff  2450 Dec  5 17:26 sign.sh
> -rw-r--r--@ 1 rohanyadav  staff   470 Dec  5 17:26 sign_deb.exp
> -rw-r--r--@ 1 rohanyadav  staff   212 Dec  5 17:26 sign_rpm.exp
> drwxr-xr-x@ 7 rohanyadav  staff   224 Dec  5 17:26 .
> -rw-r--r--@ 1 rohanyadav  staff   511 Dec  5 17:26 sign_tar.exp
> drwxr-xr-x@ 9 rohanyadav  staff   288 Dec  5 17:26 ..
> ```
> 
> Should be:
> 
> ```
> ❯ ll build/nix/
> total 40
> -rwxr-xr-x@ 1 rohanyadav  staff  1567 Nov 13 08:29 fix_archives.sh
> -rwxr-xr-x@ 1 rohanyadav  staff  2450 Dec  5 17:26 sign.sh
> -rwxr-xr-x@ 1 rohanyadav  staff   470 Dec  5 17:26 sign_deb.exp
> -rwxr-xr-x@ 1 rohanyadav  staff   212 Dec  5 17:26 sign_rpm.exp
> drwxr-xr-x@ 7 rohanyadav  staff   224 Dec  5 17:26 .
> -rwxr-xr-x@ 1 rohanyadav  staff   511 Dec  5 17:26 sign_tar.exp
> drwxr-xr-x@ 9 rohanyadav  staff   288 Dec  5 17:26 ..
> ```
> 
> Please run: `chmod 755 ./build/nix/sign_deb.exp ./build/nix/sign_rpm.exp ./build/nix/sign_tar.exp`

I ran the chmod command and pushed the updated files
For creating a pre-release should I be using dest_prefix or create it without dest_prefix?

**Comment by rajrohanyadav** [comment]
> > Did you test it by creating a prerelease? I think you'll get an error when trying to sign the packages, because the files are not executable.
> > ```
> > ❯ ll build/nix/
> > total 40
> > -rwxr-xr-x@ 1 rohanyadav  staff  1567 Nov 13 08:29 fix_archives.sh
> > -rwxr-xr-x@ 1 rohanyadav  staff  2450 Dec  5 17:26 sign.sh
> > -rw-r--r--@ 1 rohanyadav  staff   470 Dec  5 17:26 sign_deb.exp
> > -rw-r--r--@ 1 rohanyadav  staff   212 Dec  5 17:26 sign_rpm.exp
> > drwxr-xr-x@ 7 rohanyadav  staff   224 Dec  5 17:26 .
> > -rw-r--r--@ 1 rohanyadav  staff   511 Dec  5 17:26 sign_tar.exp
> > drwxr-xr-x@ 9 rohanyadav  staff   288 Dec  5 17:26 ..
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Should be:
> > ```
> > ❯ ll build/nix/
> > total 40
> > -rwxr-xr-x@ 1 rohanyadav  staff  1567 Nov 13 08:29 fix_archives.sh
> > -rwxr-xr-x@ 1 rohanyadav  staff  2450 Dec  5 17:26 sign.sh
> > -rwxr-xr-x@ 1 rohanyadav  staff   470 Dec  5 17:26 sign_deb.exp
> > -rwxr-xr-x@ 1 rohanyadav  staff   212 Dec  5 17:26 sign_rpm.exp
> > drwxr-xr-x@ 7 rohanyadav  staff   224 Dec  5 17:26 .
> > -rwxr-xr-x@ 1 rohanyadav  staff   511 Dec  5 17:26 sign_tar.exp
> > drwxr-xr-x@ 9 rohanyadav  staff   288 Dec  5 17:26 ..
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please run: `chmod 755 ./build/nix/sign_deb.exp ./build/nix/sign_rpm.exp ./build/nix/sign_tar.exp`
> 
> I ran the chmod command and pushed the updated files For creating a pre-release should I be using dest_prefix or create it without dest_prefix?

You can run with `dest_prefix`.

File: .github/workflows/on_prerelease.yaml [hunk 1]
```diff
@@ -13,4 +13,6 @@ jobs:
     with:
       tag: ${{ github.event.release.tag_name }}
       integration: "mssql"
+      upload_fips_packages: true
+      dest_prefix: "sairaj-fips-test-mssql"
```
Comment (alvarocabanas) [inline .github/workflows/on_prerelease.yaml:17]: You should remove the dest_prefix, it's meant only for testing but should be removed now we want to merge it

**Comment by sairaj18** [comment]
> You can run with `dest_prefix`.

added the test pre-release to PR description

**Comment by sairaj18** [comment]
created a new [pre-release](https://github.com/newrelic/nri-mssql/actions/runs/12366888043) and tested 

image

**Comment by rajrohanyadav** [comment]
I didn't update changelog as part of initial PR (nri-redis and other OHIs) because I thought it doesn't have any code changes. But on second thought, I believe it should be added as an `Enhancement` that we are now supporting `fips` packages. IMO, we should update it as part of the PR.

**Review by alvarocabanas — DISMISSED** [review]
LGTM

**Comment by rajrohanyadav** [comment]
> > I didn't update changelog as part of initial PR (nri-redis and other OHIs) because I thought it doesn't have any code changes. But on second thought, I believe it should be added as an `Enhancement` that we are now supporting `fips` packages. IMO, we should update it as part of the PR.
> 
> ```
> ### enhancements
> - Support fips packages for linux
> ```
> 
> This should be fine right?

I'd add something like:
```md
### dependency
- Update goreleaser to v2.4.4

### enhancements
- Add FIPS compliant packages and archives
```

**Review by rajrohanyadav — APPROVED** [review]

**Pull request merged**